docs/guides/hybrid-graphics.md

```diff
@@ -65,20 +65,64 @@ This has been tested on the MacBookPro16,1 and the MacBookPro15,1. The 15,3 and
+
+    5.  Reboot into Linux. Display brightness should be working again if it wasn't, and `glxinfo | grep "OpenGL renderer"` should show an Intel GPU. Running programs with `DRI_PRIME=1` will make them render on your AMDGPU (some things do this automatically). You will get more battery time now as your AMD GPU can be turned off when not needed.
 
 !!! note
     As macOS expects the dedicated GPU to be activated at startup, to avoid various display and GPU related problems (like not outputting anything to the display after sleep), switch to the dedicated GPU using `sudo gpu-switch -d` before booting into MacOS.
     Because of this, if you want to use Linux with the iGPU again, you'll need to re run `sudo gpu-switch -i` on your Linux install and **reboot**.
 
-# Use on Windows
+## MacBookPro16,4
+
+The AMD GPU on MacBookPro16,4 is not yet compatible with Linux. As a workaround :-
+
+### If you are able to edit your kernel command line :-
+
+1. This workaround requires GRUB as the bootloader. So make sure you install GRUB.
+
+2. Edit the command line of your boot and add the `nomodeset` kernel parameter the the command line. This will enable you to access your Linux system in safe graphics.
+
+3. Boot into Linux and install apple-os-set loader from [here](https://github.com/Redecorating/apple_set_os-loader).
+
+4. Setup this tool to allow changing the boot GPU from GRUB:
+
+    ```sh
+    git clone https://github.com/Redecorating/efi-gpu-power-prefs
+    cd efi-gpu-power-prefs
+    make #you will need efi.h to compile this, which is installed in the gnu-efi package in most distros
+    make install
+    sudo grub-mkconfig -o /boot/grub/grub.cfg
+    ```
+
+5. Install the apple-gmux driver from [here](https://github.com/Redecorating/apple-gmux-t2).
+
+6. Reboot and in the grub menu, select "Enable iGPU". Your computer will shutdown. Power it back on and boot linux. If you boot macOS, this will be reset and you'll have to redo this step.
+
+7. You can now remove the `nomodeset` parameter from your command line.
+
+### If you are unable to edit your kernel command line :-
+
+1. Boot into macOS and mount your Linux EFI partition over there. In most cases it should be `disk0s1` and can be mounted by running `sudo diskutil mount disk0s1` in the terminal. If you are using a separate EFI partition, the you can run `diskutil list` and find your partition in the output, and mount it accordingly.
+
+2. Install apple-os-set loader from [here](https://github.com/Redecorating/apple_set_os-loader) using macOS in your Linux EFI partition.
+
+3. Restart into macOS Recovery by immediately pressing and holding Command+R on startup. Open the terminal there and run `nvram fa4ce28d-b62f-4c99-9cc3-6815686e30f9:gpu-power-prefs=%01%00%00%00`.
+
+4. Restart into Linux. You should now be able to access your Linux installation.
+
+5. If your distro doesn't use GRUB, install it since further steps need GRUB.
```

Comment:
Here I'd also prefer to say you can use other bootloaders but you'll need to figure out how to configure them yourself